src/main/java/org/neo4j/gis/spatial/procedures/SpatialProcedures.java

```diff
@@ -521,21 +522,27 @@ public void removeLayer(@Name("name") String name) {
 	@Description("Adds the given node to the layer, returns the geometry-node")
 	public Stream<NodeResult> addNodeToLayer(@Name("layerName") String name, @Name("node") Node node) {
 		EditableLayer layer = getEditableLayerOrThrow(tx, spatial(), name);
-		return streamNode(layer.add(tx, node).getGeomNode());
+		Node geomNode = layer.add(tx, node).getGeomNode();
+		layer.close(tx);
```

Comment:
My idea behind the `close` was to implement `java.io.Closeable` so one can use try-with-resource. The problem here is that the index itself is not aware of the transaction, so the tx must be passed to each method as param- also to the `close` method. And so, with the additional parameter, it can no longer implement `java.io.Closeable`.

What do you think about having a Transaction-Aware Facade that can be `Closable`?

Reply:
Thanks sounds nice. Worth a try, as long as the transaction does not leak anywhere.